05_iigo/index.tex

```diff
@@ -551,8 +546,78 @@ \subsubsection{Server-side}
     \item The outcome of the action, or lack thereof, is passed to the members of the archipelago.
 \end{enumerate}
 
-\subsubsection{Legislative Branch}
-\subsubsection{Judicial Branch}
+\subsection{Legislative Branch}
+\subsection{Judicial Branch}
+The judicial branch has two parts: the Judicial branch itself, which exists on the server to perform the mechanical actions required for IIGO, and the judge role which is in the hands of the agent which is currently Judge. This agent is able to use the Judge role to make decisions which  are communicated to the serverside Judicial-branch which performs the actions.
+\subsubsection{Client-side}
+The Judge decision making functions can be divided into two categories:
+\begin{enumerate}
+    \item Judge power functions:
+    \begin{itemize}
+        \item \texttt{InspectHistory} allows the Judge to inspect the history cache and evaluate whether agents have complied with the rules or transgressed. Provided the history cache, this returns a $map[agent] \xrightarrow[]{} []transgressions$ as well as information about whether the Judge has acted on its power to inspect the history cache.
+        \item \texttt{GetRuleViolationSeverity} allows the Judge to set sanction penalties for violating specific rules. Returns a $map[ruleName] \xrightarrow[]{} penalty$. If this is not set by the Judge, the default sanction penalty is 1 unit of resources for every rule broken.
+        \item \texttt{GetSanctionThresholds} allows the Judge to set the sanction penalty thresholds for sanction tiers. Returns a $map[sanctionTier] \xrightarrow[]{} penalty$.
+        \item \texttt{GetPardonedIslands} allows the Judge to pardon islands of imposed sanctions. Provided with the list of sanctions, this returns a $map[sanctionID] \xrightarrow[]{} bool$ where a $true$ value indicates that the sanction corresponding to that $ID$ is to be removed from the list of sanctions.
+        \item \texttt{HistoricalRetributionEnabled} allows the Judge to decide whether to inspect the history and accordingly sanction unpunished transgressions from previous turns. This is a boolean value so that historical retribution can be enabled and disabled easily.
```

Comment:
```suggestion
        \item \texttt{HistoricalRetributionEnabled} allows the Judge to decide whether to inspect the history and accordingly sanction the previously unpunished transgressions from earlier turns. This is a boolean value so that historical retribution can be enabled and disabled easily.
```